com.ibm.wala.cast.js/src/main/java/com/ibm/wala/cast/js/util/FieldBasedCGUtil.java

```diff
@@ -168,6 +168,29 @@ public CallGraphResult buildScriptDirCG(
         loaders, scripts.toArray(new Module[0]), builderType, monitor, supportFullPointerAnalysis);
   }
 
+  /**
+   * Construct a bounded field-based call graph using all the {@code .js} files appearing in
+   * scriptDir or any of its sub-directories
+   */
+  public CallGraphResult buildScriptDirBoundedCG(
+      Path scriptDir, IProgressMonitor monitor, boolean supportFullPointerAnalysis, Integer bound)
+      throws WalaException, CancelException, IOException {
+    JavaScriptLoaderFactory loaders = new JavaScriptLoaderFactory(translatorFactory);
+    List<Path> jsFiles =
+        Files.walk(scriptDir)
+            .filter(p -> p.toString().toLowerCase().endsWith(".js"))
+            .collect(Collectors.toList());
+    List<Module> scripts = new ArrayList<>();
+    // we can't do this loop as a map() operation on the previous stream because toURL() throws
+    // a checked exception
+    for (Path p : jsFiles) {
+      scripts.add(new SourceURLModule(p.toUri().toURL()));
+    }
+    scripts.add(JSCallGraphUtil.getPrologueFile("prologue.js"));
```

Comment:
This code is shared with `buildScriptDirCG()` I think.  Can we extract it to a shared function, something like `List<Module> findScriptsInDir(Path scriptDir)`, and call it from both places?